python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -655,8 +655,31 @@ def open(self, data):
         :param dict data: Message payload.
         """
         try:
-            # Retrieve filepath.
-            filepath = data.get("filepath")
+            # Retrieve filepath. We should always get something in the payload, but if
+            # we didn't for some reason, passing on an empty string will ensure that
+            # a sane exception is raised later on when the existence of the path is
+            # checked.
+            filepath = data.get("filepath", "")
+            local_storages = data.get("local_storages")
+
+            # Logging here is for debugging purposes. We have a situation where some
+            # clients are reporting errors when opening files from Shotgun, and the
+            # error implies that we're getting a null value for the file path passed
+            # down from the web app. There are reasonable situations where this might
+            # happen, but in the cases we're attempting to debug it shouldn't be the 
+            # case.
+            if filepath is None:
```

Comment:
It might be worth to `logger.debug` the path in the `else:` clause of this `if` statement. This way it puts the next `log.debug` call more in perspective instead of simply knowing it's the "current" one.

Reply:
👍